frontend/src/pages/BatchSend.vue

```diff
@@ -0,0 +1,538 @@
+        const token: TokenInfoExtended | null = Sends.value[i].token;
+        if (!token) throw new Error('Send.please-complete-form');
+        const { address: tokenAddress, decimals } = token;
+        console.log('tokenAddres is ', tokenAddress);
+        const tokenAmount = parseUnits(Sends.value[i].amount, decimals);
+
+        if (tokenAddress)
+          newSends.push({ token: tokenAddress, amount: tokenAmount, address: Sends.value[i].receiver as string });
+      }
+      console.log('batchSend that is sent over to umbra.ts is', newSends);
+
+      // Send with Umbra
+      const { tx } = await umbra.value.batchSend(signer.value, newSends, {
+        // advanced: shouldUseNormalPubKey.value,
+        // // When attempting to sendMax, we override the gasPrice to use the price
+        // // from the sweepETH function that estimated the tokenAmount. That function
+        // // calculates the tokenAmount based on a low (but reasonable) estimate of
+        // // the gasPrice. Wallets are configured to choose a high-probability
+        // // gasPrice for new transactions -- which is to say: a fairly high gasPrice.
+        // // So if we don't specify the gasPrice here, the wallet will almost
+        // // certainly choose a higher gasPrice. And since:
+        // //   token amount = (account balance) - (expected gas costs)
+        // // when the wallet increases gas costs on us, we end up in a situation where:
+        // //   account balance < (token amount) + (wallet-chosen gas costs)
+        // // So the wallet will reject the transaction on grounds that the account
+        // // doesn't have enough funds.
+        // gasPrice: sendMax.value && sendingNativeToken ? sendMaxGasPrice : undefined,
+        // gasLimit: sendMax.value && sendingNativeToken ? sendMaxGasLimit : undefined,
+      });
+      void txNotify(tx.hash, ethersProvider);
+      await tx.wait();
+      resetForm();
+    } finally {
+      isSending.value = false;
+      showAdvancedSendWarning.value = false;
+    }
+  }
+
+  function resetForm() {
+    recipientId.value = undefined;
+    token.value = undefined;
+    humanAmount.value = undefined;
+    sendFormRef.value?.resetValidation();
+  }
+
+  return {
+    acknowledgeSendRisk,
+    advancedAcknowledged,
+    advancedMode,
+    balances,
+    connectRedirectTo,
+    chainId,
+    currentChain,
+    humanAmount,
+    humanAmountBaseInputRef,
+    humanToll,
+    humanTotalAmount,
+    isSending,
+    isValidForm,
+    isValidId,
+    addFields,
+    isValidRecipientId,
+    isValidTokenAmount,
+    NATIVE_TOKEN,
+    onFormSubmit,
+    paymentLinkParams,
+    recipientId,
+    recipientIdBaseInputRef,
+    sendAdvancedButton,
+    sendFormRef,
+    sendMax,
+    showAdvancedSendWarning,
+    showAdvancedWarning,
+    token,
+    tokenBaseSelectRef,
+    tokenList,
+    toll,
+    useNormalPubKey,
+    userAddress,
+    Sends,
+  };
+}
+
+export default defineComponent({
+  name: 'BatchSend',
```

Comment:
Let's also rename the file to `AccountBatchSend`. The naming convention that I like for components that we use in Umbra is namespacing them. So we have `AccountSend` and `AccountReceive` pages because send and receive are actions an account can take. Then we have `AccountReceiveTable` because that's a table meant to be used with `AccountReceive`, etc.
```suggestion
  name: 'PageBatchSend',
```

Reply:
Renamed in [914b7cc](https://github.com/ScopeLift/umbra-protocol/pull/491/commits/914b7cc6b1d4d871ceb4913d27baf46eb97610f1)